lib/src/rules/avoid_field_initializers_in_const_classes.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r'Avoid field initializers in const classes.';
+
+const _details = r'''
+
+**AVOID** field initializers in const classes.
+
+Instead of `final x = const expr;`, you should write `get x => const expr;` and
+not allocate a useless field.
```

Comment:
Perhaps point out that this is true for the VM, but not for code that will be compiled to JS.